# Week 2 Checkpoint: First Data Profile
**BAN 6003 Data Management and Analytics Integration**

Complete this notebook using the `employee_data.csv` file in the `data/` folder.

This is the independent checkpoint for Week 2. Use the functions practiced in the guided lab. You are not expected to clean the data yet. Your goal is to profile the data and report what you notice.


## Instructions

1. Run each cell from top to bottom.
2. Fill in all cells marked `# Your code here`.
3. Write short answers in the markdown response areas.
4. Save, commit, and push your notebook to GitHub.

You may use the Week 2 guided lab as a reference.


## Task 0: Import Pandas and Load the Data
Run the cell below.


In [1]:
import pandas as pd

df = pd.read_csv("../data/employee_data.csv")
df.head()


,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
0,101,John,Smith,4/15/22,Marketing,85000,34.0,True
1,102,Jane,Doe,8/20/21,Sales,92000,45.0,True
2,103,Peter,Jones,1/2/22,HR,71000,29.0,True
3,104,Mary,Johnson,11/10/23,Engineering,115000,38.0,True
4,105,Mike,Brown,5/30/20,sales,89000,NaN,False


## Task 1: Basic Structure
Show the basic structure of the DataFrame.

Your code should show:
- the shape of the DataFrame;
- the column names;
- the first five rows;
- the last five rows.


In [2]:
print("Shape:", df.shape)
print("Columns:", df.columns.tolist())
print(df.head())
print(df.tail())


Shape: (13, 8)
Columns: ['employee_id', 'first_name', 'last_name', 'hire_date', 'department', 'salary', 'age', 'full_time']
  employee_id first_name last_name hire_date   department  salary   age  \
0         101       John     Smith   4/15/22    Marketing   85000  34.0   
1         102       Jane       Doe   8/20/21        Sales   92000  45.0   
2         103      Peter     Jones    1/2/22           HR   71000  29.0   
3         104       Mary   Johnson  11/10/23  Engineering  115000  38.0   
4         105       Mike     Brown   5/30/20        sales   89000   NaN   

  full_time  
0      True  
1      True  
2      True  
3      True  
4     False  
   employee_id first_name last_name hire_date   department    salary    age  \
8          102      Emily     White   1/15/24        Sales     91000   31.0   
9          110      David    Miller   7/22/19           HR     68000   58.0   
10         111      Sarah     Moore   3/12/23    Marketing  $91,000    28.0   
11       error      Kevin

### Task 1 Written Response
Write 1–2 sentences describing the size and general structure of the dataset.

**Your response:**  
The dataset contains 13 rows and 8 columns, with each row representing an employee and columns capturing information such as employee ID, name, hire date, department, salary, age, and full time status.


## Task 2: Data Types and Non-Missing Counts
Use Pandas to inspect data types and non-missing counts.

Your code should use:
- `info()`
- `dtypes`


In [4]:
# Your code here
df.info()
df.dtypes

<class 'pandas.DataFrame'>
RangeIndex: 13 entries, 0 to 12
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   employee_id  13 non-null     str    
 1   first_name   13 non-null     str    
 2   last_name    13 non-null     str    
 3   hire_date    12 non-null     str    
 4   department   13 non-null     str    
 5   salary       12 non-null     str    
 6   age          12 non-null     float64
 7   full_time    12 non-null     object 
dtypes: float64(1), object(1), str(6)
memory usage: 964.0+ bytes


employee_id        str
first_name         str
last_name          str
hire_date          str
department         str
salary             str
age            float64
full_time       object
dtype: object

### Task 2 Written Response
Which columns, if any, have data types that might need to be investigated later? For example, is a number or date stored as text/object?

**Your response:**  
The employee_id, hire_date, and salary columns may need further investigation because they are stored as text even though they represent IDs, dates, and numerical amounts. The full_time column is also stored as an object rather than a Boolean and may need investigation because it contains a missing value.


## Task 3: Missing Values
Check missing values for all columns.

Your code should use `isna().sum()`.

Optional: also calculate missing percentages.


In [5]:
# Your code here
df.isna().sum()

employee_id    0
first_name     0
last_name      0
hire_date      1
department     0
salary         1
age            1
full_time      1
dtype: int64

### Task 3 Written Response
Which columns have missing values? Which missing values seem most important to investigate later?

**Your response:**  
The hire_date, salary, age, and full_time columns each have one missing value. The missing salary and age values may be especially important to investigate because they could affect financial and employee-level analysis.


## Task 4: Numerical Summary
Use `.describe()` to summarize numerical columns.

Then choose one numerical column and calculate at least two individual summary statistics such as mean, median, min, max, or standard deviation.


In [7]:
# Your code here
print(df.describe())

print("Mean age:", df["age"].mean())
print("Max age:", df["age"].max())

              age
count   12.000000
mean   118.583333
std    277.411889
min     28.000000
25%     32.500000
50%     36.500000
75%     46.750000
max    999.000000
Mean age: 118.58333333333333
Max age: 999.0


### Task 4 Written Response
What do you notice from the numerical summaries? Mention at least one range, unusual value, or pattern.

**Your response:**  
The age values range from 28 to 999, with a mean of approximately 118.6 years. The maximum value of 999 is unusually high and should be investigated before using age for analysis.


## Task 5: Categorical Summary
Use `.value_counts()` to summarize at least one categorical column.

Start with `department` if it exists in the dataset. Then choose one additional categorical column if available.


In [8]:
# Your code here
df["department"].value_counts()

department
Sales          4
Marketing      3
Engineering    3
HR             2
sales          1
Name: count, dtype: int64

### Task 5 Written Response
What are the most common categories? Do you notice anything that might need standardization or further investigation later?

**Your response:**  
The most common department is Sales with 4 records, followed by Marketing and Engineering with 3 records each. The department values "Sales" and "sales" use inconsistent capitalization and should be standardized before grouping or aggregating by department.


## Task 6: Duplicate Checks
Check for duplicate records and duplicate employee IDs.

Your code should check:
- fully duplicated rows;
- duplicate `employee_id` values;
- the most common `employee_id` values using `value_counts()`.


In [9]:
# Your code here
full_duplicate_rows = df.duplicated().sum()
print("Fully duplicated rows:", full_duplicate_rows)

employee_id_duplicates = df["employee_id"].duplicated().sum()
print("Duplicated employee_id count:", employee_id_duplicates)

df["employee_id"].value_counts()

Fully duplicated rows: 0
Duplicated employee_id count: 1


employee_id
102      2
101      1
103      1
104      1
105      1
106      1
107      1
108      1
110      1
111      1
error    1
113      1
Name: count, dtype: int64

### Task 6 Written Response
Are there duplicate employee IDs? If yes, what might that mean if `employee_id` is supposed to uniquely identify each employee?

**Your response:**  
Yes. Employee ID 102 appears twice, resulting in one duplicated employee_id value. If employee_id is supposed to uniquely identify each employee, this may indicate a duplicate record or data entry issue and should be investigated before analysis.


## Task 7: Overall Data Quality Observations
Write 3–5 bullet points summarizing the condition of the dataset.

Focus on what you found from profiling. Do **not** clean the data yet.

Examples:
- The dataset has ___ rows and ___ columns.
- The ___ column has missing values.
- The ___ column may have an incorrect data type.
- The ___ column has possible duplicate IDs.
- The ___ column has a range from ___ to ___.


**Your observations:**

1. The dataset contains 13 rows and 8 columns, capturing employee-level information such as employee ID, name, hire date, department, salary, age, and full-time status.
2. The hire_date, salary, age, and full_time columns each contain one missing value.
3. The employee_id, hire_date, salary, and full_time columns have data types or values that may need further investigation.
4. The department column has inconsistent capitalization, with "Sales" and "sales" treated as separate categories.
5. Employee ID 102 appears twice, and one employee_id value is listed as "error" rather than a numeric ID. These issues should be investigated if employee_id is intended to uniquely identify each employee.


## Submission Checklist
Before submitting, make sure:

- [ ] All code cells run without errors.
- [ ] You completed all tasks.
- [ ] You wrote short responses in the markdown cells.
- [ ] You saved the notebook.
- [ ] You committed and pushed your work to GitHub.

Use the terminal:

```bash
git add .
git commit -m "Completed Week 2 data profile checkpoint"
git push
```
